src/main/java/com/yannbriancon/interceptor/HibernateQueryInterceptor.java

```diff
@@ -70,16 +68,29 @@ public void startQueryCount() {
         threadQueryCount.set(0L);
     }
 
+    /**
+     * Start or reset the query count to 0 for the considered thread.
+     * Counts only if given filter matches.
+     *
+     * @param filter string filter predicate
+     */
+    public void startQueryCount(Predicate<String> filter) {
+        this.filter = filter;
+        startQueryCount();
```

Comment:
startQueryCount() should be the one calling startQueryCount(null) and not the contrary.
Here calling startQueryCount() does not reset the filter.

Reply:
done